lib/routable_state.js

```diff
@@ -0,0 +1,161 @@
+(function() {
+  "use strict";
+
+  var State = require('./state'), router = require('./router'), util = require('./util');
+
+  // Public: Provides a `State` subclass that integrates seemlessly with the `Router` singleton.
+  // `RoutableState` instances have methods for defining routes that map to that particular state.
+  //
+  // Examples
+  //
+  // var statechart = RoutableState.define(function() {
+  //   this.state('foosIndex', function() {
+  //     this.route('/foos');
+  //   });
+  //
+  //   this.state('foosShow', function() {
+  //     this.route('/foos/:id');
+  //   });
+  // });
+  function RoutableState() {
+    State.apply(this, arguments);
+  }
+
+  util.assign(RoutableState, State);
+
+  RoutableState.prototype = Object.create(State.prototype);
+  RoutableState.prototype.constructor = RoutableState;
+
+  // Public: Define a route for the current state. When this state is entered the route will be set
+  // as the current route on the `router` singleton. Also, when the user changes the URL to match
+  // the route, a state transition is automatically triggered to this state.
+  //
+  // Routes can be defined on the state that they map to directly or at some parent state. Defining
+  // multiple routes at a parent state gives you the ability to control the order in which the
+  // states are defined and therefore searched for matches.
+  //
+  // pattern - A route pattern (see the docs for `Router#define`).
+  // state   - Either a string containing a state path or an actual `State` object.
```

Comment:
I can't remember why I decided to allow it to accept `State` objects, but I think you're right, it doesn't seem to be necessary. I'll remove it.